core-bundle/src/Picker/PickerConfig.php

```diff
@@ -111,9 +111,9 @@ public static function urlDecode(string $data): self
             $decoded = $uncompressed;
         }
 
-        $json = @json_decode($decoded, true);
-
-        if (null === $json) {
+        try {
+            $json = json_decode($decoded, true, 512, JSON_THROW_ON_ERROR);
+        } catch (\JsonException) {
             throw new \InvalidArgumentException('Invalid JSON data');
         }
```

Comment:
Changing the exception type would be a BC break, especially because the method has an `@throws` annotation. I have therefore merged your second suggestion in e5255709da7dc9ee31945e118046dd15bde79857.